x-pack/plugins/monitoring/server/telemetry_collection/get_logstash_stats.ts

```diff
@@ -324,37 +394,42 @@ export async function fetchLogstashState(
   ephemeralIds: string[],
   start: string,
   end: string,
-  { page = 0, ...options }: { page?: number } & LogstashProcessOptions
+  { page = 0, ...options }: { page?: number } & LogstashProcessOptions,
+  isSelfMonitoring: boolean = false
 ): Promise<void> {
+  const stateField = isSelfMonitoring ? 'logstash_state' : 'logstash.node.state';
+  const filterPath: string[] = [
+    `hits.hits._source.${stateField}.pipeline.batch_size`,
+    `hits.hits._source.${stateField}.pipeline.workers`,
+    `hits.hits._source.${stateField}.pipeline.representation.graph.vertices`,
+    `hits.hits._source.type`,
+  ];
+
   const params: estypes.SearchRequest = {
     index: INDEX_PATTERN_LOGSTASH,
     ignore_unavailable: true,
-    filter_path: [
-      'hits.hits._source.logstash_state.pipeline.batch_size',
-      'hits.hits._source.logstash_state.pipeline.workers',
-      'hits.hits._source.logstash_state.pipeline.representation.graph.vertices.config_name',
-      'hits.hits._source.logstash_state.pipeline.representation.graph.vertices.plugin_type',
-      'hits.hits._source.logstash_state.pipeline.representation.graph.vertices.meta.source.protocol',
-      'hits.hits._source.logstash_state.pipeline.representation.graph.vertices',
-      'hits.hits._source.type',
-    ],
+    filter_path: filterPath,
     body: {
       query: createQuery({
-        start,
-        end,
+        // intentionally not using start and end periods as we need recent node state info to fill plugin usages
+        // especially with metricbeat monitoring
```

Comment:
review note: metricbeat occasionally sends the _state_ event(s). Id we apply `{start, end}` period, we don't get any result. If we don't have result, we fail filling the pipeline definition information (like which plugin is used or queue type, etc...)

Reply:
> isn't periodic report, data will be emitted when pipeline changes (pipeline reload, LS restart, etc..).

Two questions from this: 
1. Could it happen that the ILM deletes it after 6 days and no periodic report is emitted? => Missing telemetry
2. If it isn't periodic, could it happen that the data we report may be outdated? Or we're not intentionally updating it because we're only reporting when anything contained in the reported info changes?